source/patterns/@aws-solutions-constructs/aws-lambda-kinesisfirehose/test/aws-lambda-kinesisfirehose.test.ts

```diff
@@ -0,0 +1,208 @@
+/**
+ *  Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+// Imports
+import '@aws-cdk/assert/jest';
+import { Construct } from 'constructs';
+import * as cdk from 'aws-cdk-lib';
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+import { KinesisFirehoseToS3 } from '@aws-solutions-constructs/aws-kinesisfirehose-s3';
+import { LambdaToKinesisFirehose } from '../lib';
+import * as defaults from '@aws-solutions-constructs/core';
+
+test('Test that properties are set correctly', () => {
+  // Stack
+  const stack = new cdk.Stack();
+  const destination = GetTestDestination(stack, 'test-destination');
+
+  const construct = new LambdaToKinesisFirehose(stack, 'test-target', {
+    existingKinesisFirehose: destination.kinesisFirehose,
+    lambdaFunctionProps: {
+      runtime: lambda.Runtime.NODEJS_18_X,
+      handler: "index.handler",
+      code: lambda.Code.fromAsset(`${__dirname}/lambda`),
+    }
+  });
+
+  expect(construct.kinesisFirehose).toBeDefined();
+  expect(construct.lambdaFunction).toBeDefined();
+  expect(construct.vpc).not.toBeDefined();
+});
+
+test('Test existing function is utilzed correctly', () => {
+  // Stack
+  const stack = new cdk.Stack();
+  const destination = GetTestDestination(stack, 'test-destination');
+  const testName = 'test-name';
+  const existingFunction = new lambda.Function(stack, 'test-function', {
+    runtime: lambda.Runtime.NODEJS_18_X,
+    handler: "index.handler",
+    code: lambda.Code.fromAsset(`${__dirname}/lambda`),
+    functionName: testName
+  });
+
+  new LambdaToKinesisFirehose(stack, 'test-target', {
+    existingKinesisFirehose: destination.kinesisFirehose,
+    existingLambdaObj: existingFunction
+  });
+
+  expect(stack).toHaveResource('AWS::Lambda::Function', {
+    FunctionName: testName
+  });
+
+  // This is 2 because there's a lambda function in the custom resource to
+  // delete all the objects when cleaning up the s3 bucket in kinesisfirehose-s3
```

Comment:
In going through the PR top-down, I'm surprised to see references to kinesisfirehose-s3. Is this comment accurate? Its not obvious to me why/how we are using it.

Reply:
Now I see why below in the definition of the `GetTestDestination` function.